projects/models.nix

```
@@ -0,0 +1,128 @@
+{
+  lib,
+  pkgs,
+  sources,
+}:
+let
+  yants = import sources.yants { };
+
+  inherit (yants)
+    string
+    list
+    option
+    attrs
+    enum
+    either
+    struct
+    drv
+    path
+    restrict
+    ;
+
+  programType = struct "program" {
+    name = string;
+    source = drv;
```

We decided programs should also be NixOS modules (`programs.<symbolic application name>`), so the actual sources would be an implementation detail we later think about whether and how expose it in the UI. So arguably that could just as well be

```suggestion
    module = defun [attrs attrs];
```